user/tests/integration/slo/connect_time/connect_time.cpp

```diff
@@ -0,0 +1,117 @@
+#include "application.h"
+#include "test.h"
+
+#if Wiring_Cellular
+#define Network Cellular
+#elif Wiring_Wifi
+#define Network WiFi
+#else
+#error "Unsupported platform"
+#endif
+
+namespace {
+
+const auto TIMEOUT = 10 * 60 * 1000; // Default timeout for waitFor()
+
+struct State {
+    system_tick_t networkColdConnectTimeFromStartup;
+    system_tick_t networkWarmConnectTimeFromStartup;
+    system_tick_t cloudFullHandshakeDuration;
+    system_tick_t cloudSessionResumeDuration;
+};
+
+retained State s = {};
+
+system_tick_t globalInitTimeFromStartup = 0;
+system_tick_t setupTimeFromStartup = 0;
+system_tick_t loopTimeFromStartup = 0;
+bool loopCalled = false;
+
+system_tick_t testAppInitDuration = 0;
+system_tick_t testAppSetupDuration = 0;
+
+size_t serializeStateAsJson(char* buf, size_t size) {
+    JSONBufferWriter w(buf, size);
+    w.beginObject();
+    w.name("network_cold_connect_time_from_startup").value((unsigned)s.networkColdConnectTimeFromStartup);
+    w.name("network_warm_connect_time_from_startup").value((unsigned)s.networkWarmConnectTimeFromStartup);
+    w.name("cloud_full_handshake_duration").value((unsigned)s.cloudFullHandshakeDuration);
+    w.name("cloud_session_resume_duration").value((unsigned)s.cloudSessionResumeDuration);
+    w.endObject();
+    return w.dataSize();
+}
+
+} // namespace
+
+STARTUP({
+    globalInitTimeFromStartup = millis();
+    testAppInit();
+    testAppInitDuration = millis() - globalInitTimeFromStartup;
+});
+
+void setup() {
+    setupTimeFromStartup = millis();
+    testAppSetup();
+    testAppSetupDuration = millis() - setupTimeFromStartup;
+    setupTimeFromStartup -= testAppInitDuration;
+}
+
+void loop() {
+    if (!loopCalled) {
+        loopTimeFromStartup = millis() - testAppInitDuration - testAppSetupDuration;
+        loopCalled = true;
+    }
+    testAppLoop();
+}
+
+test(network_connect_cold) {
+    Network.off(); // Make sure the network interface is tuned off
```

Comment:
Changed to a function so that's it's less confusing:
```cpp
inline NetworkClass& network() {
#if Wiring_Cellular
    return Cellular;
#elif Wiring_Wifi
    return WiFi;
#else
#error "Unsupported platform"
#endif
}
```